### DISCUSSION TITLE
feature suggestion: dispatching workflows from pull requests

### DISCUSSION THREAD

**Discussion by OmarTawfik** [community/community#62190] [Actions]
### Select Topic Area

Product Feedback

## Problem

Large repositories/projects often need to automate certain reviewer/author actions that can be done on PRs (like rebasing/applying certain rules/tests/etc...). They do that by implementing GitHub apps that respond to commands in comments on these PRs, like `@bot do something`.

To make this easier, GitHub also provides UI to trigger workflows from the workflow page by copying the branch name/commit hash to https://github.com/{org}/{repo}/actions/workflows/{name}.yml and launching it.

<img width="354" alt="image" src="https://github.com/community/community/assets/15987992/9eba4033-6588-475c-a3e2-36c6ef416213">

This is a couple of extra steps that can be error prone (copying the wrong branch, or forgetting to change from the default `main`) when juggling multiple PRs. Additionally, it causes a much bigger issue for discovery, where developers don't even know what workflows they can run on such PRs. In mono-repos (or teams who have dedicated infra people), it becomes much harder to keep up with any such changes.

## Suggestion

Allow authors to specify in `.github/workflows/{name}.yml` which PRs it can apply to. It will allow triggering these actions from the PR itself, passing the correct git ref:

<img width="369" alt="image" src="https://github.com/community/community/assets/15987992/7d8a8b55-fae4-4ec2-afd4-73ce2aa4c17b">

**Comment by GhostOf0days** [comment]
? This is already possible. See [events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows). Here is an example workflow file located in .github/workflows:

```
name: Lint
on: [push, pull_request]

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: bahmutov/npm-install@v1
      - name: Check TypeScript
        run: npx tsc
        shell: bash
      - name: Run ESLint
        run: npm run lint
        shell: bash

**Reply by OmarTawfik** [reply]
I'm not sure I'm following. The event you mentioned triggers the workflow when a PR is opened/updated. The suggestion is about being able to trigger workflows manually, **after it is created**, from the web UI on the PR page. Can you please clarify?

**Reply by GhostOf0days** [reply]
Oh sorry. I misread your post. You're right. Manually triggering it from a PR is not possible. The only bypass is adding a new label and triggering it upon adding a new label. There is no manual way to trigger workflows on a PR. See this [Stack Overflow discussion](https://stackoverflow.com/questions/63288356/how-to-link-a-github-actions-manual-run-to-a-pr) for more context.
